ledger/src/blockstore.rs

```diff
@@ -1292,9 +1292,39 @@ impl Blockstore {
         };
 
         if shred.data_header.size == 0 {
+            let leader_pubkey = leader_schedule
+                .map(|leader_schedule| leader_schedule.slot_leader_at(slot, None))
+                .unwrap_or(None);
```

Comment:
instead of `map(...).unwrap_or(None)` you can use `.and_then`:
https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then

similarly for the other case below

Reply:
Thanks for the tip! Didn't know about `.and_then()`